### PR TITLE
Change action isDisabled to accept promise

### PR DESCRIPTION
**Pull request by JSReds**
Change-type: patch
Signed-off-by: Andrea Rosci <andrear@balena.io>

Screenshot 2022-12-21 at 12 25 06

Screenshot 2022-12-21 at 17 36 40

### CODE REVIEW

File: src/AutoUI/Actions/Update.tsx [hunk 1]
```diff
@@ -87,38 +142,60 @@ export const Update = <T extends AutoUIBaseResource<T>>({
 			</Box>
 		);
 	}
+	const groupedActions = React.useMemo(
+		() => groupBy(updateActions, (action) => action.section),
+		[updateActions],
+	);
+	const actionHandlers = React.useMemo(
```
Comment (thgreasi) [inline src/AutoUI/Actions/Update.tsx:149]: These hook now need to go above the early return `if (!updateActions || updateActions.length < 1) { return null; }` that we have earlier in this file.

File: src/AutoUI/Actions/Update.tsx [hunk 2]
```diff
@@ -87,38 +142,60 @@ export const Update = <T extends AutoUIBaseResource<T>>({
...
-					}),
-				tooltip:
-					typeof disabledReason === 'string' ? disabledReason : undefined,
-				disabled: !!disabledReason,
-				danger: action.isDangerous,
-			};
-		}),
+					return {
+						content: (
+							<DropdownItemContent<any>
```
Comment (thgreasi) [inline src/AutoUI/Actions/Update.tsx:163]: Cant' this be
```suggestion
							<DropdownItemContent<T>
```
?
Not a blocker though.

File: src/AutoUI/utils.ts [hunk 3]
```diff
@@ -93,7 +93,7 @@ export const getTagsDisabledReason = <T extends AutoUIBaseResource<T>>(
 		return t('info.edit_tag_no_permissions', { resource: 'item' });
 	}
 
-	return false;
+	return undefined;
```
Comment (thgreasi) [inline src/AutoUI/utils.ts:96]: Would it work if we just removed this ?
```suggestion
```

**Review by thgreasi — COMMENTED** [review]
Just move the `useMemo()`s above the early return and LGTM :+1:

File: src/AutoUI/Actions/Update.tsx (hunk 1 above)
Comment (JSReds) [reply src/AutoUI/Actions/Update.tsx:149]: ufff you are right, nice catch 🙏

File: src/AutoUI/Actions/Update.tsx (hunk 2 above)
Comment (JSReds) [reply src/AutoUI/Actions/Update.tsx:163]: I forgot to change `DropDownItemContentProps` after some tests... thank you :)

File: src/AutoUI/utils.ts (hunk 3 above)
Comment (JSReds) [reply src/AutoUI/utils.ts:96]: :+1:

**Review by thgreasi — COMMENTED** [review]
LGTM, change it to a minor and get it in :tada:

**Comment by JSReds** [comment]
@balena-ci I self-certify!

**Pull request merged**